src/llm/llmServices/deepSeek/deepSeekService.ts

```diff
@@ -0,0 +1,157 @@
+    DeepSeekUserModelParams,
+    DeepSeekModelParams,
+    DeepSeekService,
+    DeepSeekGeneratedProof,
+    DeepSeekServiceInternal
+> {
+    readonly serviceName = "DeepSeekService";
+    protected readonly internal = new DeepSeekServiceInternal(
+        this,
+        this.eventLogger,
+        this.generationsLoggerBuilder
+    );
+    protected readonly modelParamsResolver = new DeepSeekModelParamsResolver();
+
+    static readonly baseApiUrl = "https://api.deepseek.com/v1";
+}
+
+export class DeepSeekGeneratedProof extends GeneratedProofImpl<
+    DeepSeekModelParams,
+    DeepSeekService,
+    DeepSeekGeneratedProof,
+    DeepSeekServiceInternal
+> {
+    constructor(
+        rawProof: GeneratedRawContentItem,
+        proofGenerationContext: ProofGenerationContext,
+        modelParams: OpenAiModelParams,
+        llmServiceInternal: DeepSeekServiceInternal,
+        previousProofVersions?: ProofVersion[]
+    ) {
+        super(
+            rawProof,
+            proofGenerationContext,
+            modelParams,
+            llmServiceInternal,
+            previousProofVersions
+        );
+    }
+}
+
+class DeepSeekServiceInternal extends LLMServiceInternal<
+    DeepSeekModelParams,
+    DeepSeekService,
+    DeepSeekGeneratedProof,
+    DeepSeekServiceInternal
+> {
+    constructGeneratedProof(
+        rawProof: GeneratedRawContentItem,
+        proofGenerationContext: ProofGenerationContext,
+        modelParams: OpenAiModelParams,
+        previousProofVersions?: ProofVersion[] | undefined
+    ): DeepSeekGeneratedProof {
+        return new DeepSeekGeneratedProof(
+            rawProof,
+            proofGenerationContext,
+            modelParams,
+            this,
+            previousProofVersions
+        );
+    }
+
+    async generateFromChatImpl(
+        analyzedChat: AnalyzedChatHistory,
+        params: OpenAiModelParams,
+        choices: number
+    ): Promise<GeneratedRawContent> {
+        LLMServiceInternal.validateChoices(choices);
+
+        const openai = new OpenAI({
+            apiKey: params.apiKey,
+            baseURL: DeepSeekService.baseApiUrl,
+        });
+        const formattedChat = this.formatChatHistory(analyzedChat.chat, params);
+        this.logDebug.event("Completion requested", {
+            history: formattedChat,
+        });
+
+        try {
+            const completion = await openai.chat.completions.create({
+                messages: formattedChat,
+                model: params.modelName,
+                n: choices,
+                temperature: params.temperature,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                max_tokens: params.maxTokensToGenerate,
+            });
+            const rawContentItems = completion.choices.map((choice) => {
+                const content = choice.message.content;
+                if (content === null) {
+                    throw Error("response message content is null");
```

Comment:
After the last refactor this line should look like `illegalState("response message content is null");` as it is already in `OpenAIService` 🫡

Reply:
Kein Problem